tests/geomultiscale-mapping/RadialGeoMultiscale.cpp

```diff
@@ -18,61 +18,66 @@ BOOST_AUTO_TEST_CASE(RadialGeoMultiscale)
 
   Participant cplInterface(context.name, context.config(), 0, 1);
   if (context.isNamed("SolverOne")) {
-    auto     meshName = "MeshOne";
-    Vector3d vec1     = Vector3d::Constant(0.0);
-    auto     vid      = cplInterface.setMeshVertex(meshName, vec1);
-    auto     dataAID  = "DataOne";
-    auto     dataBID  = "DataTwo";
+    Vector3d            coordOneA{0.0, 0.0, 0.0};
+    Vector3d            coordOneB{0.0, 0.0, 1.0};
+    Vector3d            coordOneC{0.0, 0.0, 2.0};
+    std::vector<double> values;
+    const unsigned int  nCoords = 3;
+    for (unsigned int i = 0; i < nCoords; ++i) {
+      values.emplace_back(std::pow(i + 1, 2));
+    }
+    auto             meshOneID = "MeshOne";
+    std::vector<int> ids;
+    ids.emplace_back(cplInterface.setMeshVertex(meshOneID, coordOneA));
+    ids.emplace_back(cplInterface.setMeshVertex(meshOneID, coordOneB));
+    ids.emplace_back(cplInterface.setMeshVertex(meshOneID, coordOneC));
+
+    auto dataAID = "DataOne";
 
-    double valueDataB = 1.0;
     cplInterface.initialize();
     double maxDt = cplInterface.getMaxTimeStepSize();
-    cplInterface.readData(meshName, dataBID, {&vid, 1}, maxDt, {&valueDataB, 1});
-    BOOST_TEST(valueDataB == 1.0);
-
-    while (cplInterface.isCouplingOngoing()) {
+    BOOST_TEST(cplInterface.isCouplingOngoing());
 
-      double data[] = {1.0};
-      cplInterface.writeData(meshName, dataAID, {&vid, 1}, data);
+    cplInterface.writeData(meshOneID, dataAID, ids, values);
+    cplInterface.advance(maxDt);
 
-      cplInterface.advance(maxDt);
-      maxDt = cplInterface.getMaxTimeStepSize();
-
-      cplInterface.readData(meshName, dataBID, {&vid, 1}, maxDt, {&valueDataB, 1});
-      BOOST_TEST(valueDataB == 1.0);
-    }
     cplInterface.finalize();
 
+    std::cout << "This is done 1 \n";
+
   } else {
     BOOST_TEST(context.isNamed("SolverTwo"));
-    auto     meshName = "MeshTwo";
-    Vector3d vec2     = Vector3d::Constant(1.0);
-    auto     vid      = cplInterface.setMeshVertex(meshName, vec2);
+    auto meshTwoID = "MeshTwo";
+
+    Vector3d coordTwoA{0.5, 0.0, 0.0};
+    Vector3d coordTwoB{0.5, 0.0, 1.0};
+    Vector3d coordTwoC{0.5, 0.0, 2.0};
```

Comment:
The mesh of SolverTwo seems to be the same as the mesh of SolverOne, offset by x=0.5.
Maybe add two more points in the middle, to ensure that there is always really a mapping going on and not just a copy.

Reply:
I kept 3 points, but I moved them in y-direction as well, such that it is not just an offset to the original mesh but three distinct points.